daemon/image_delete.go

```diff
@@ -132,15 +146,16 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 				if first {
 					return err
 				}
-
 			}
-
 		}
 	}
 	return nil
 }
 
 func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
+	directlyUsedBy := []string{}
```

Comment:
We divide transitive links and primary links with these, I'm assuming. Is there a way we can do this without considering transitivity? Either the image is used or it is not. Aren't we just considering in-links to decide if the image can be deleted?